magellan-compose/src/main/java/com/wealthfront/magellan/compose/core/Navigator.kt

```diff
@@ -26,7 +26,7 @@ import java.util.LinkedList
 
 class Navigator(private val container: (Context) -> ScreenContainer?) : LifecycleAwareComponent() {
 
-  private val backStack: Deque<Screen> = LinkedList()
+  private val backStack: Deque<Screen<*>> = LinkedList()
```

Comment:
I don't think our navigator should be using `Screen`s directly, instead using navigables